Trello Task 3 - GET /api/topics

husainbismil opened the pull request.

husainbismil committed:
in progress

husainbismil committed:
Merge branch 'main' into 3_get-api-topics

husainbismil committed:
done 3. get api/topics

k-huggs commented on __tests__/app.test.js:
```diff
@@ -0,0 +1,40 @@
+            return request(app).get("/api/topics").expect(200).then((response) => {
+                const topics = response.body;
+
+                expect(topics).toBeInstanceOf(Array);
```
No need to check that this will be an instance of an array

k-huggs commented on app/app.js:
```diff
@@ -0,0 +1,25 @@
+
+
+// Parse JSON by default
+app.use(express.json());
```
You're not using this so please delete until it's being utilised

k-huggs commented on app/app.js:
```diff
@@ -0,0 +1,25 @@
+
+// Error Handling
+
+app.all('/*', (req, res) => {
```
You've put this in, however, there are no tests for a 404 in your test file. Please add the test prior to implementation

k-huggs commented on controller/controller.js:
```diff
@@ -0,0 +1,17 @@
+exports.getApiTopics = (request, response) => {
+    m.selectTopics().then((queryResult) => {
+        const responseArray = queryResult.rows;
+        response.status(200).send(responseArray);
```
You don't want to be sending this back as an array, it should be on a object with the key being what specifically it is

k-huggs commented on controller/controller.js:
```diff
@@ -0,0 +1,17 @@
+
+// /api/topics Endpoint
+exports.getApiTopics = (request, response) => {
+    m.selectTopics().then((queryResult) => {
```
semantic variable naming please, query result can be anything!

k-huggs commented on model/model.js:
```diff
@@ -0,0 +1,9 @@
+// c = controller, m = model, v = view, db = database
+const db = require("../db/connection.js");
+const c = require("../controller/controller.js");
```
You don't need to have the controller required into your model

k-huggs commented on package.json:
```diff
@@ -21,14 +21,17 @@
-    "pg": "^8.7.3"
+    "dotenv": "^16.0.3",
+    "express": "^4.18.2",
+    "lodash": "^4.17.21",
```
lodash has been installed but, what is it purpose currently?

k-huggs reviewed (commented):
A few changes to make, have a look at the comments and resubmit :)

husainbismil committed:
removed Array check

removed line 25:                 expect(topics).toBeInstanceOf(Array);

husainbismil committed:
removed app use express json for now

husainbismil committed:
removed non TDD code, see description

removed error handling that wasn't being tested for in the test suite yet

husainbismil committed:
making variable names more semantic, see desc

const queryResult -> selectTopicsQueryResult

husainbismil committed:
changing format of response, see desc

instead of responding with an array, response changed to an object, with keyval array, and key describing the contents of the keyval.

husainbismil committed:
removed 'require controller' from model

removed 'require controller' from model as its not needed there

husainbismil committed:
removed lodash as its not being used

husainbismil commented:
> A few changes to make, have a look at the comments and resubmit :)

changes made, thank you!

husainbismil committed:
fixing test suite after previous changes

k-huggs commented on controller/controller.js:
```diff
@@ -0,0 +1,17 @@
+// /api/topics Endpoint
+exports.getApiTopics = (request, response) => {
+    m.selectTopics().then((selectTopicsQueryResult) => {
+        const responseObject = {topics: selectTopicsQueryResult.rows};
```
The controller should just been handling the req and res. Any other manipulation should be performed in your model

k-huggs reviewed (approved):
One small change then merge 👍

husainbismil committed:
adding in kyle fixes
